torch/distributed/_tensor/api.py

```diff
@@ -592,25 +592,43 @@ def distribute_tensor(
             f"`placements` must have the same length as `device_mesh.ndim`! "
             f"Found placements length: {len(placements)}, and device_mesh.ndim: {device_mesh.ndim}."
         )
+
+    parent_mesh: Optional[DeviceMesh] = None
     if isinstance(tensor, DTensor):
-        # if the tensor is already a DTensor, we need to check:
-        # 1. if the we can further shard this DTensor if the two device mesh belong to
-        #   the same parenet mesh and further sharding is possible.
-        # 2. check if device mesh and placements are the same
-        if tensor.device_mesh != device_mesh:
+        if tensor.device_mesh == device_mesh:
+            if tensor.placements != tuple(placements):
+                raise ValueError(
+                    f"Cannot distribute a DTensor with placements {tensor.placements} "
+                    f"to a different placements {placements}. do you want to call "
+                    f"`redistribute` instead?"
+                )
+            return tensor
+
+        parent_mesh = _mesh_resources.get_parent_mesh(tensor.device_mesh)
+        if parent_mesh is not None and _mesh_resources.get_parent_mesh(device_mesh):
+            if parent_mesh.ndim != 2:
+                raise ValueError(
+                    "Currently, distribute_tensor only supports 2D parent device mesh."
+                )
+            # TODO: these checks need to be extended once we support nD parent
+            # device mesh.
+            if (
+                _mesh_resources.get_parent_mesh_dim(tensor.device_mesh) == 0
+                and next(iter(tensor.placements)).is_shard()
+                and placements[0].is_shard()
+            ):
+                raise ValueError(
```

Comment:
We could support multi Shard placements on multiple mesh dim if:

The input dtensor's mesh is in the left index compare to the passed in `device_mesh` in the parent_mesh mesh_dims (In this case, we can further shard the DTensor on the same dimension)